src/long.js

```diff
@@ -1086,6 +1086,47 @@ LongPrototype.modulo = function modulo(divisor) {
     return this.sub(this.div(divisor).mul(divisor));
 };
 
+/**
+ * Returns a deep copy of this Long.
+ * @this {!Long}
+ * @returns {!Long} Deep copy of this Long
+ */
+LongPrototype.clone = function clone() {
+    return new Long(this.low, this.high, this.unsigned);
+};
+
+/**
+ * Returns this Long to given 32bit integer power.
+ * @this {!Long}
+ * @param {!Long|number} exp Integer power (32bit)
+ * @returns {!Long} This Long to given integer power
+ */
+LongPrototype.power = function power(exp) {
+    var a = this;
+    if (a.eq(Long.ONE)) return a;
+    if (isLong(exp)) exp = exp.toInt();
```

Comment:
Would the preference be to keep exponent as a Long and manipulate it as a 64 bit value? I figured it would be faster to use 32 bit math as exponent values larger than 64 would overflow on bases beyond 0 and 1. So at the time the thought was speed rather than safety. I could certainly put the 64 bit exp math back into it if that is desired. Just let me know.

Reply:
Or is the concern that it could quietly give bad values in the overflow case? Does it need some kind of range check or overflow error of some kind? One could throw if base > 1 and exp > 64 but that would only be a broad check and there is plenty of combos that would still overflow. Not sure if there is a way to detect overflows during the multiplication?